src/cudadecoder/batched-threaded-cuda-decoder.cc

```diff
@@ -0,0 +1,637 @@
+// cudadecoder/batched-threaded-cuda-decoder.cc
+/*
+ * Copyright (c) 2017, NVIDIA CORPORATION.  All rights reserved.
+ * Authors:  Hugo Braun, Justin Luitjens, Ryan Leary
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define SLEEP_BACKOFF_NS 500
+#define SLEEP_BACKOFF_S ((double)SLEEP_BACKOFF_NS/1e9)
+#if HAVE_CUDA == 1
+#include "cudadecoder/batched-threaded-cuda-decoder.h"
+#include "base/kaldi-utils.h"
+#include <nvToolsExt.h>
+
+namespace kaldi {
+namespace CudaDecode {
+
+void BatchedThreadedCudaDecoder::Initialize(
+    const fst::Fst<fst::StdArc> &decode_fst, const nnet3::AmNnetSimple &am_nnet,
+    const TransitionModel &trans_model) {
+  KALDI_LOG << "BatchedThreadedCudaDecoder Initialize with "
```

Comment:
you should also print out the num_worker_threads to be consistent

Reply:
thanks! also added the batch size